hare/broker.go

```diff
@@ -138,7 +100,9 @@ func (b *Broker) HandleMessage(ctx context.Context, _ p2p.Peer, msg []byte) pubs
 
 func (b *Broker) handleMessage(ctx context.Context, msg []byte) error {
 	h := types.CalcMessageHash12(msg, pubsub.HareProtocol)
-	logger := b.WithContext(ctx).WithFields(log.Stringer("latest_layer", b.getLatestLayer()), h)
+	b.mu.RLock()
+	logger := b.WithContext(ctx).WithFields(log.Stringer("latest_layer", b.latestLayer), h)
+	b.mu.RUnlock()
```

Comment:
doing this is error prone. i'd much rather restore getLatestLayer with the pattern
```
lock
defer unlock
```
whenever possible.